### PR TITLE
Create a yum repo based on the image manifest content

### PR DESCRIPTION
**Pull request by lmilbaum**
Creating a yum repo, uploading it to S3 bucket and testing that it is a valid yum repo.
Signed-off-by: Liora Milbaum <lmilbaum@redhat.com>

### CODE REVIEW

File: tests/ci/yumrepo.py
Comment (RajeshKumarSrivastava) [inline tests/ci/yumrepo.py]: "el8" should be part of the nvr read from "c8s-image-manifest.txt" . It misses those. I will need to regenerate that list again without stripping "el8" so that you don't have to add it here

File: .github/workflows/ateam.yml
Comment (juanje) [inline .github/workflows/ateam.yml]: Maybe it'd be easier to use centos, or fedora, container for the jobs and move the logic from the Dockerfile to here.
Here we set what container we use:
https://github.com/osbuild/automotive-ci/blob/main/.github/workflows/ateam.yml#L17

As you're not running the others jobs for your tests, you should be able to change the container. Also, I think you can use different containers for different jobs:
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainerimage

File: tests/ci/yumrepo.py
Comment (lmilbaum) [reply tests/ci/yumrepo.py]: Removed _el8_

File: .github/workflows/ateam.yml
Comment (lmilbaum) [reply .github/workflows/ateam.yml]: I've take your advice. Thanks.

File: .github/workflows/ateam.yml
Comment (juanje) [inline .github/workflows/ateam.yml:34]: This looks much better :smile: 
Be aware that all this is running inside the container, which means that the repo is created inside the container. The next step (**Image build**) will run in the bare-metal machine, so it won't see this container or the repo just created here.
For the **Image Builder** (`osbuild`) to see the repo you'll need to do one of two things:
- Rsync the repo directory to the bare-metal machine (`${{ secrets.HOST }}`)
- Upload to S3 and use it from there.

I like the second option more.
Here is an axample from `osbuild`: https://github.com/osbuild/osbuild-composer/blob/main/schutzbot/mockbuild.sh#L111
Here is also an article about it. In case it helps: https://reece.tech/posts/hosting-centos-7-and-8-yum-repositories-in-s3/

Comment (lmilbaum) [reply .github/workflows/ateam.yml:34]: Thanks. Pushed the changes.

File: .github/workflows/a-team.yml
Comment (hubeadmin) [inline .github/workflows/a-team.yml]: I propose we use a more verbose secret, the local env for this action run is fine, but we want to differentiate between the various types of buckets we will have on AWS in our GH secret store, i.e for your use it would be `AWS_BUCKET: ${{ secrets.AWS_BUCKET_YUM }}` to indicate you want to use the yum repo bucket, and another secret for Guy's needs e.g `${{ secrets.AWS_BUCKET_IMG}}`, Which are secrets that I would hope terraform can somehow populate (securely) , I'm still figuring it out on the terraform end though.

Comment (lmilbaum) [reply .github/workflows/a-team.yml]: Good idea.

**Review by hubeadmin — APPROVED** [review]
/lgtm !

**Comment by lmilbaum** [comment]
@RajeshKumarSrivastava would #109 unblock the broken CI of this PR?

**Comment by RajeshKumarSrivastava** [comment]
> @RajeshKumarSrivastava would #109 unblock the broken CI of this PR?

I don't think the CI is broken because of the package not downloaded. I see this `ERROR:root:Exception Failed to upload ./repo/libgcab1-1.1-1.el8.aarch64.rpm to ***/1012485273/libgcab1-1.1-1.el8.aarch64.rpm: An error occurred (NoSuchBucket) when calling the PutObject operation: The specified bucket does not exist encountered while uploading file./repo/libgcab1-1.1-1.el8.aarch64.rpm` . If that's already resolved, once I merge the packages, it should solve the package not found problem

**Comment by lmilbaum** [comment]
> > @RajeshKumarSrivastava would #109 unblock the broken CI of this PR?
> 
> I don't think the CI is broken because of the package not downloaded. I see this `ERROR:root:Exception Failed to upload ./repo/libgcab1-1.1-1.el8.aarch64.rpm to ***/1012485273/libgcab1-1.1-1.el8.aarch64.rpm: An error occurred (NoSuchBucket) when calling the PutObject operation: The specified bucket does not exist encountered while uploading file./repo/libgcab1-1.1-1.el8.aarch64.rpm` . If that's already resolved, once I merge the packages, it should solve the package not found problem

Yeah. This is the second issue which causing the CI to be broken. @HubertStefanski can you please create the bucket manually in the A-Team account to unblock the broken CI?

**Comment by lmilbaum** [comment]
@RajeshKumarSrivastava I've rebase the PR. Still hitting some packages which are not found. Can you please take a look?

**Comment by RajeshKumarSrivastava** [comment]
> @RajeshKumarSrivastava I've rebase the PR. Still hitting some packages which are not found. Can you please take a look?

yes, I see that, will have a look

File: .github/workflows/a-team.yml
Comment (juanje) [inline .github/workflows/a-team.yml]: @lioramilbaum  Not sure about the motive for this change, but at the official documentation it uses single quotes:
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-ignoring-paths

Maybe using double quotes might create unexpected behaviors, like expanding the glob before to pass it to the `path-ignore`directive. I don't know. I'd rather leave it as single quotes, just in case.

Comment (lmilbaum) [reply .github/workflows/a-team.yml]: This change is done automatically by a vscode format plugin.  I will remove it.

**Comment by juanje** [comment]
@lioramilbaum do you have the repo URL or definition, so we could create a `.repo` file and tests locally in a container or something? Just to be sure it's a valid yum repository.

**Review by brejoc — APPROVED** [review]
lgtm!

**Comment by lmilbaum** [comment]
> @lioramilbaum do you have the repo URL or definition, so we could create a `.repo` file and tests locally in a container or something? Just to be sure it's a valid yum repository.

It is located in the S3 bucket. I can add a test which verifies it is a valid yum repo. WDYT?

File: ci/create_yum_repo/yumrepo.j2
Comment (juanje) [inline ci/create_yum_repo/yumrepo.j2:2]: Maybe we could put here a sorted name so it's easier to use at `dnf/yum`.

The name at the `[]` is usually the long name and at the field `name` the sort one. (I think that) usually without spaces.

Comment (lmilbaum) [reply ci/create_yum_repo/yumrepo.j2:2]: I have looked at some of the repos on my system. Here is one 
[hashicorp-test]
name=Hashicorp Test - $basearch
baseurl=https://rpm.releases.hashicorp.com/RHEL/7/$basearch/test
enabled=0
gpgcheck=1
gpgkey=https://rpm.releases.hashicorp.com/gpg`

Comment (juanje) [reply ci/create_yum_repo/yumrepo.j2:2]: That's true, it's exactly the opposite of what I told you :sweat_smile:

**Review by juanje — APPROVED** [review]
LGTM!

File: ci/create_yum_repo/main.py
Comment (brejoc) [inline ci/create_yum_repo/main.py]: Just a little nitpick. While I think using a string shouldn't be a problem in this case, a list would be cleaner.

File: ci/create_yum_repo/utils.py
Comment (brejoc) [inline ci/create_yum_repo/utils.py]: Do we really need a full shell here?

File: ci/create_yum_repo/main.py
Comment (brejoc) [inline ci/create_yum_repo/main.py]: Can't we check the log lvl first, so that we wouldn't even open the file otherwise?

**Review by brejoc — CHANGES_REQUESTED** [review]
Some nitpicks!

**Review by RajeshKumarSrivastava — COMMENTED** [review]
do you need ci/__init__.py?

File: ci/create_yum_repo/main.py
Comment (RajeshKumarSrivastava) [reply ci/create_yum_repo/main.py]: one of the idea for log levels is to have filter post. We shouldn't log on condition

Comment (lmilbaum) [reply ci/create_yum_repo/main.py]: what doesn't it mean - "to have filer post" ?

**Review by juanje — APPROVED** [review]
LGTM!

**Review by guy9050 — APPROVED** [review]

File: ci/create_yum_repo/main.py
Comment (RajeshKumarSrivastava) [reply ci/create_yum_repo/main.py]: typo , "filter". We shouldn't have conditional logs. This filtering should be done post by a log aggregator.

Comment (lmilbaum) [reply ci/create_yum_repo/main.py]: ack

**Review by hubeadmin — APPROVED** [review]
/lgtm

**Comment by lmilbaum** [comment]
@brejoc please review/approve?

**Review by RajeshKumarSrivastava — APPROVED** [review]

**Review by brejoc — APPROVED** [review]
lgtm!

**Pull request merged**
